extension/src/SymbolReference/SymbolReferenceCache.ts

```diff
@@ -1,48 +1,43 @@
-import { AppPackage } from "./types/AppPackage";
+import { AppPackage, AppIdentifier } from "./types/AppPackage";
 
-const cachedAppPackages: AppPackage[] = [];
+export class SymbolReferenceCache {
+  private cache: Map<string, AppPackage>;
 
-export function addAppPackageToCache(appPackage: AppPackage): void {
-  if (appPackageInCache(appPackage)) {
-    // app already in cache
-    return;
+  public get size(): number {
+    return this.cache.size;
   }
 
-  const appToCache = appPackage;
-  appToCache.symbolReference = undefined; // Free up unnecessary memory allocation
-  cachedAppPackages.push(appToCache);
-  cachedAppPackages.sort((a, b) => {
-    return a.sort(b);
-  });
-}
+  constructor() {
+    this.cache = new Map();
+  }
 
-export function getAppPackageFromCache(
-  name: string,
-  publisher: string,
-  version: string
-): AppPackage {
-  const cachedAppPackage = cachedAppPackages.filter(
-    (p) => p.name === name && p.publisher === publisher && p.version === version
-  )[0];
-  return cachedAppPackage;
-}
+  static key(appId: AppIdentifier): string {
+    return `${appId.name}-${appId.publisher}-${appId.version}`;
+  }
 
-function appPackageInCache(appPackage: AppPackage): boolean {
-  return (
-    cachedAppPackages.filter((p) => p.packageId === appPackage.packageId)
-      .length > 0
-  );
-}
+  get(appId: AppIdentifier): AppPackage | undefined {
+    return this.cache.get(SymbolReferenceCache.key(appId));
+  }
+
+  set(appPackage: AppPackage): void {
+    if (this.isCached(appPackage)) {
+      return;
+    }
+    const appToCache = appPackage;
+    this.cache.set(SymbolReferenceCache.key(appToCache), appToCache);
+  }
```

Comment:
@jwikman  note that the `appPackage.symbolReference` is not cleared in this function anymore. Since I couldn't properly clone an object I felt it was best to remove that to avoid future side effects.

Reply:
It's now cleared in `getObjectsFromAppFile` as we discussed.

Reply:
> It's now cleared in `getObjectsFromAppFile` as we discussed.

I think it looks good now 👍